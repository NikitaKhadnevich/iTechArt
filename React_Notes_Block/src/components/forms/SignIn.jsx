import * as React from 'react';
import Avatar from '@mui/material/Avatar';
import TextField from '@mui/material/TextField';
import Grid from '@mui/material/Grid';
import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
import Typography from '@mui/material/Typography';
import {
  SignContainer,
  SignBox,
  FormBox,
  FormGrid,
  FormButton,
} from './styled';

export default function SignIn() {
  const handleSubmit = (event) => {
    event.preventDefault();
    const data = new FormData(event.currentTarget);
    // eslint-disable-next-line no-console
    console.log({
      email: data.get('email'),
      password: data.get('password'),
    });
  };

  return (
    <SignContainer component='main'>
      <SignBox>
        <Avatar sx={{ m: 1, bgcolor: 'red' }}>
          <LockOutlinedIcon />
        </Avatar>
        <Typography component='h1' variant='h5'>
          Sign In
        </Typography>
        <FormBox
          component='form'
          noValidate
          onSubmit={handleSubmit}
          sx={{ mt: 3 }}
        >
          <FormGrid container spacing={2}>
            <Grid item xs={12}>
              <TextField
                id='email'
                name='email'
                required
                fullWidth
                label='Email Address'
                autoComplete='email'
                autoFocus
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                id='password'
                name='password'
                required
                fullWidth
                label='Password'
                type='password'
                autoComplete='new-password'
              />
            </Grid>
          </FormGrid>
          <FormButton type='submit' variant='contained' sx={{ mt: 2, mb: 2 }}>
            Sign Up
          </FormButton>
        </FormBox>
      </SignBox>
    </SignContainer>
  );
}
