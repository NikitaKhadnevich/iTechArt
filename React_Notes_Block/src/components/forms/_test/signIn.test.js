/* eslint-disable no-unused-vars */
/* eslint-disable no-shadow */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import * as reactRedux from 'react-redux';
import { configure, shallow, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import SignIn from '../SignIn';

configure({ adapter: new Adapter() }); // ES6 modules

describe('SignIn', () => {
  const formik = {
    values: { email: 'email', password: 'password' },
    errors: false,
    touched: false,
    handleChange: jest.fn(),
    isValid: false,
    setFieldTouched: jest.fn(),
    handleBlur: jest.fn(),
    handleSubmit: jest.fn(),
  };
  const signInLoaded = {
    formik,
    isFetching: false,
    isLoading: false,
  };

  const signInLoading = {
    formik,
    isFetching: true,
    isLoading: true,
  };

  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
  });

  it('Is it SignIn Exist and Loaded ', () => {
    const signInWrapper = shallow(<SignIn {...signInLoaded} />);
    expect(signInWrapper).toMatchSnapshot();
    expect(signInWrapper).toHaveLength(1);
  });

  it('should components data are loading/Fetching', () => {
    const loadingWrapper = shallow(<SignIn {...signInLoading} />);
    expect(loadingWrapper.find('Spinner')).toHaveLength(1);
    expect(loadingWrapper).toMatchSnapshot();
  });
});
