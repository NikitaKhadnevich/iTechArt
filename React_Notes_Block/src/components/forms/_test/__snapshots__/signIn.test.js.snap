// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Is it SignIn Exist and Loaded  1`] = `
<Styled(ForwardRef(Container))
  component="main"
>
  <Styled(ForwardRef(Box))>
    <SignInInfo
      welcomeSignIn="Hello, Let's auth"
    />
    <Styled(ForwardRef(Box))
      component="form"
      noValidate={true}
      onSubmit={[MockFunction]}
      sx={
        Object {
          "mt": 3,
        }
      }
    >
      <Styled(ForwardRef(Grid))
        container={true}
        spacing={2}
      >
        <ForwardRef(Grid)
          item={true}
          xs={12}
        >
          <SignEmail
            email="email"
            errors={false}
            handleBlur={[MockFunction]}
            onChangeAccum={[Function]}
            touched={false}
          />
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          xs={12}
        >
          <SignPassword
            errors={false}
            handleBlur={[MockFunction]}
            onChangeAccum={[Function]}
            password="password"
            touched={false}
          />
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(Button))
            color="secondary"
            disabled={true}
            fullWidth={true}
            type="submit"
            variant="outlined"
          >
            Sign In
          </WithStyles(ForwardRef(Button))>
        </ForwardRef(Grid)>
      </Styled(ForwardRef(Grid))>
    </Styled(ForwardRef(Box))>
  </Styled(ForwardRef(Box))>
</Styled(ForwardRef(Container))>
`;

exports[`SignIn should components data are loading/Fetching 1`] = `
<Styled(ForwardRef(Container))
  component="main"
>
  <Spinner />
</Styled(ForwardRef(Container))>
`;
